src/main/java/org/example/commands/DeleteNodeCommand.java

```diff
@@ -3,42 +3,64 @@
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
-import java.util.Scanner;
 import org.example.exceptions.InvalidArgument;
 import org.example.model.Node;
+import org.example.services.ConsoleInputServiceInterface;
 
 public class DeleteNodeCommand implements CommandInterface {
+  private final ConsoleInputServiceInterface consoleInputService;
+
+  public DeleteNodeCommand(final ConsoleInputServiceInterface consoleInputService) {
+    this.consoleInputService = consoleInputService;
+  }
 
   @Override
-  public void execute(Map<String, Node> nodeDependencies) {
-    Scanner scanner = new Scanner(System.in);
+  public void execute(final Map<String, Node> nodeDependencies) {
     System.out.println("Enter the node id you want to delete with all its dependencies");
 
-    String nodeId = scanner.nextLine();
+    final String nodeId = consoleInputService.inputNodeId();
+
+    validateNodeId(nodeId);
+    deleteNodeWithDependencies(nodeDependencies, nodeId);
+  }
+
+  private void validateNodeId(final String nodeId) {
```

Comment:
this type of function is used in multiple place, can you put in one common place and reuse it

Reply:
Created a utils package for these type of functions